### ISSUE TITLE
Error installing floatview in Docker

### ISSUE THREAD

**Issue by LukaPitamic** [denphi/jupyterlab-floatview#5]
Cannot install FloatView in docker.
Attached is the whole error log. Outside docker it installs flawlessly. 

<details>
  <summary>Error LOG</summary>
  
## # pip install floatview
Collecting floatview
  Using cached floatview-0.2.0-py2.py3-none-any.whl (5.8 MB)
Collecting glueviz>=0.15.2
  Using cached glueviz-0.15.2-py2.py3-none-any.whl (2.7 kB)
Collecting networkx>=2.2
  Using cached networkx-2.4-py3-none-any.whl (1.6 MB)
Collecting scikit-learn>=0.19.0
  Using cached scikit_learn-0.22.2.post1-cp36-cp36m-manylinux1_x86_64.whl (7.1 MB)
Requirement already satisfied: ipywidgets>=7.5.0 in /usr/local/lib/python3.6/site-packages (from floatview) (7.5.1)
Requirement already satisfied: plotly>=4.0.0 in /usr/local/lib/python3.6/site-packages (from floatview) (4.6.0)
Collecting colorlover>=0.3.0
  Using cached colorlover-0.3.0-py3-none-any.whl (8.9 kB)
Collecting glue-vispy-viewers>=0.12.2
  Using cached glue_vispy_viewers-0.12.2-py2.py3-none-any.whl (950 kB)
Collecting glue-core>=0.15.5
  Using cached glue_core-0.15.6-py2.py3-none-any.whl (1.5 MB)
Requirement already satisfied: decorator>=4.3.0 in /usr/local/lib/python3.6/site-packages (from networkx>=2.2->floatview) (4.4.2)
Requirement already satisfied: scipy>=0.17.0 in /usr/local/lib/python3.6/site-packages (from scikit-learn>=0.19.0->floatview) (1.4.1)
Requirement already satisfied: numpy>=1.11.0 in /usr/local/lib/python3.6/site-packages (from scikit-learn>=0.19.0->floatview) (1.18.2)
Collecting joblib>=0.11
  Using cached joblib-0.14.1-py2.py3-none-any.whl (294 kB)
Requirement already satisfied: nbformat>=4.2.0 in /usr/local/lib/python3.6/site-packages (from ipywidgets>=7.5.0->floatview) (5.0.4)
Requirement already satisfied: ipython>=4.0.0; python_version >= "3.3" in /usr/local/lib/python3.6/site-packages (from ipywidgets>=7.5.0->floatview) (7.13.0)
Requirement already satisfied: ipykernel>=4.5.1 in /usr/local/lib/python3.6/site-packages (from ipywidgets>=7.5.0->floatview) (5.2.0)
Requirement already satisfied: widgetsnbextension~=3.5.0 in /usr/local/lib/python3.6/site-packages (from ipywidgets>=7.5.0->floatview) (3.5.1)
Requirement already satisfied: traitlets>=4.3.1 in /usr/local/lib/python3.6/site-packages (from ipywidgets>=7.5.0->floatview) (4.3.3)
Requirement already satisfied: six in /usr/local/lib/python3.6/site-packages (from plotly>=4.0.0->floatview) (1.14.0)
Requirement already satisfied: retrying>=1.3.3 in /usr/local/lib/python3.6/site-packages (from plotly>=4.0.0->floatview) (1.3.3)
Collecting astropy>=1.1
  Using cached astropy-4.0.1.post1-cp36-cp36m-manylinux1_x86_64.whl (6.5 MB)
Collecting qtpy
  Using cached QtPy-1.9.0-py2.py3-none-any.whl (54 kB)
Requirement already satisfied: pillow in /usr/local/lib/python3.6/site-packages (from glue-vispy-viewers>=0.12.2->glueviz>=0.15.2->floatview) (7.1.1)
Requirement already satisfied: matplotlib in /usr/local/lib/python3.6/site-packages (from glue-vispy-viewers>=0.12.2->glueviz>=0.15.2->floatview) (3.2.1)
Collecting pyopengl
  Using cached PyOpenGL-3.1.5-py3-none-any.whl (2.4 MB)
Collecting bottleneck>=1.2
  Using cached Bottleneck-1.3.2.tar.gz (88 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting h5py>=2.4
  Using cached h5py-2.10.0-cp36-cp36m-manylinux1_x86_64.whl (2.9 MB)
Collecting mpl-scatter-density>=0.5
  Using cached mpl_scatter_density-0.6-py2.py3-none-any.whl (655 kB)
Requirement already satisfied: pandas>=0.14 in /usr/local/lib/python3.6/site-packages (from glue-core>=0.15.5->glueviz>=0.15.2->floatview) (0.25.3)
Requirement already satisfied: setuptools>=1.0 in /usr/local/lib/python3.6/site-packages (from glue-core>=0.15.5->glueviz>=0.15.2->floatview) (46.1.3.post20200330)
Collecting qtconsole
  Using cached qtconsole-4.7.2-py2.py3-none-any.whl (117 kB)
Processing /root/.cache/pip/wheels/09/84/74/d2b4feb9ac9488bc83c475cb2cbe8e8b7d9cea8320d32f3787/dill-0.3.1.1-py3-none-any.whl
Collecting xlrd>=1.0
  Using cached xlrd-1.2.0-py2.py3-none-any.whl (103 kB)
Requirement already satisfied: jsonschema!=2.5.0,>=2.4 in /usr/local/lib/python3.6/site-packages (from nbformat>=4.2.0->ipywidgets>=7.5.0->floatview) (3.2.0)
Requirement already satisfied: ipython-genutils in /usr/local/lib/python3.6/site-packages (from nbformat>=4.2.0->ipywidgets>=7.5.0->floatview) (0.2.0)
Requirement already satisfied: jupyter-core in /usr/local/lib/python3.6/site-packages (from nbformat>=4.2.0->ipywidgets>=7.5.0->floatview) (4.6.3)
Requirement already satisfied: backcall in /usr/local/lib/python3.6/site-packages (from ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (0.1.0)
Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.6/site-packages (from ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (3.0.5)
Requirement already satisfied: pygments in /usr/local/lib/python3.6/site-packages (from ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (2.6.1)
Requirement already satisfied: jedi>=0.10 in /usr/local/lib/python3.6/site-packages (from ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (0.16.0)
Requirement already satisfied: pickleshare in /usr/local/lib/python3.6/site-packages (from ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (0.7.5)
Requirement already satisfied: pexpect; sys_platform != "win32" in /usr/local/lib/python3.6/site-packages (from ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (4.8.0)
Requirement already satisfied: jupyter-client in /usr/local/lib/python3.6/site-packages (from ipykernel>=4.5.1->ipywidgets>=7.5.0->floatview) (6.1.2)
Requirement already satisfied: tornado>=4.2 in /usr/local/lib/python3.6/site-packages (from ipykernel>=4.5.1->ipywidgets>=7.5.0->floatview) (6.0.4)
Requirement already satisfied: notebook>=4.4.1 in /usr/local/lib/python3.6/site-packages (from widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (6.0.3)
Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.6/site-packages (from matplotlib->glue-vispy-viewers>=0.12.2->glueviz>=0.15.2->floatview) (1.2.0)
Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.6/site-packages (from matplotlib->glue-vispy-viewers>=0.12.2->glueviz>=0.15.2->floatview) (0.10.0)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /usr/local/lib/python3.6/site-packages (from matplotlib->glue-vispy-viewers>=0.12.2->glueviz>=0.15.2->floatview) (2.4.7)
Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.6/site-packages (from matplotlib->glue-vispy-viewers>=0.12.2->glueviz>=0.15.2->floatview) (2.8.1)
Collecting fast-histogram>=0.3
  Using cached fast_histogram-0.8-cp36-cp36m-manylinux2010_x86_64.whl (37 kB)
Requirement already satisfied: pytz>=2017.2 in /usr/local/lib/python3.6/site-packages (from pandas>=0.14->glue-core>=0.15.5->glueviz>=0.15.2->floatview) (2019.3)
Requirement already satisfied: pyzmq>=17.1 in /usr/local/lib/python3.6/site-packages (from qtconsole->glue-core>=0.15.5->glueviz>=0.15.2->floatview) (19.0.0)
Requirement already satisfied: pyrsistent>=0.14.0 in /usr/local/lib/python3.6/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets>=7.5.0->floatview) (0.16.0)
Requirement already satisfied: importlib-metadata; python_version < "3.8" in /usr/local/lib/python3.6/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets>=7.5.0->floatview) (1.6.0)
Requirement already satisfied: attrs>=17.4.0 in /usr/local/lib/python3.6/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets>=7.5.0->floatview) (19.3.0)
Requirement already satisfied: wcwidth in /usr/local/lib/python3.6/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (0.1.9)
Requirement already satisfied: parso>=0.5.2 in /usr/local/lib/python3.6/site-packages (from jedi>=0.10->ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (0.6.2)
Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.6/site-packages (from pexpect; sys_platform != "win32"->ipython>=4.0.0; python_version >= "3.3"->ipywidgets>=7.5.0->floatview) (0.6.0)
Requirement already satisfied: prometheus-client in /usr/local/lib/python3.6/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (0.7.1)
Requirement already satisfied: terminado>=0.8.1 in /usr/local/lib/python3.6/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (0.8.3)
Requirement already satisfied: jinja2 in /usr/local/lib/python3.6/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (2.11.1)
Requirement already satisfied: Send2Trash in /usr/local/lib/python3.6/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (1.5.0)
Requirement already satisfied: nbconvert in /usr/local/lib/python3.6/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (5.6.1)
Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.6/site-packages (from importlib-metadata; python_version < "3.8"->jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets>=7.5.0->floatview) (3.1.0)
Requirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.6/site-packages (from jinja2->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (1.1.1)
Requirement already satisfied: mistune<2,>=0.8.1 in /usr/local/lib/python3.6/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (0.8.4)
Requirement already satisfied: entrypoints>=0.2.2 in /usr/local/lib/python3.6/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (0.3)
Requirement already satisfied: pandocfilters>=1.4.1 in /usr/local/lib/python3.6/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (1.4.2)
Requirement already satisfied: bleach in /usr/local/lib/python3.6/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (3.1.4)
Requirement already satisfied: defusedxml in /usr/local/lib/python3.6/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (0.6.0)
Requirement already satisfied: testpath in /usr/local/lib/python3.6/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (0.4.4)
Requirement already satisfied: webencodings in /usr/local/lib/python3.6/site-packages (from bleach->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.5.0->floatview) (0.5.1)
Building wheels for collected packages: bottleneck
  Building wheel for bottleneck (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python /usr/local/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmpsc3667t3
       cwd: /tmp/pip-install-qpe6heyv/bottleneck
  Complete output (122 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/bottleneck
  copying bottleneck/_version.py -> build/lib.linux-x86_64-3.6/bottleneck
  copying bottleneck/_pytesttester.py -> build/lib.linux-x86_64-3.6/bottleneck
  copying bottleneck/__init__.py -> build/lib.linux-x86_64-3.6/bottleneck
  creating build/lib.linux-x86_64-3.6/bottleneck/slow
  copying bottleneck/slow/nonreduce_axis.py -> build/lib.linux-x86_64-3.6/bottleneck/slow
  copying bottleneck/slow/reduce.py -> build/lib.linux-x86_64-3.6/bottleneck/slow
  copying bottleneck/slow/nonreduce.py -> build/lib.linux-x86_64-3.6/bottleneck/slow
  copying bottleneck/slow/move.py -> build/lib.linux-x86_64-3.6/bottleneck/slow
  copying bottleneck/slow/__init__.py -> build/lib.linux-x86_64-3.6/bottleneck/slow
  creating build/lib.linux-x86_64-3.6/bottleneck/benchmark
  copying bottleneck/benchmark/bench.py -> build/lib.linux-x86_64-3.6/bottleneck/benchmark
  copying bottleneck/benchmark/autotimeit.py -> build/lib.linux-x86_64-3.6/bottleneck/benchmark
  copying bottleneck/benchmark/bench_detailed.py -> build/lib.linux-x86_64-3.6/bottleneck/benchmark
  copying bottleneck/benchmark/__init__.py -> build/lib.linux-x86_64-3.6/bottleneck/benchmark
  creating build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/nonreduce_axis_test.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/list_input_test.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/reduce_test.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/memory_test.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/scalar_input_test.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/move_test.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/util.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/nonreduce_test.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/input_modification_test.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  copying bottleneck/tests/__init__.py -> build/lib.linux-x86_64-3.6/bottleneck/tests
  creating build/lib.linux-x86_64-3.6/bottleneck/src
  copying bottleneck/src/bn_config.py -> build/lib.linux-x86_64-3.6/bottleneck/src
  copying bottleneck/src/bn_template.py -> build/lib.linux-x86_64-3.6/bottleneck/src
  copying bottleneck/src/__init__.py -> build/lib.linux-x86_64-3.6/bottleneck/src
  UPDATING build/lib.linux-x86_64-3.6/bottleneck/_version.py
  set build/lib.linux-x86_64-3.6/bottleneck/_version.py to '1.3.2'
  running build_ext
  running config
  compiling '_configtest.c':
  
  #pragma GCC diagnostic error "-Wattributes"
  
  int __attribute__((optimize("O3"))) have_attribute_optimize_opt_3(void*);
  
  int main(void)
  {
      return 0;
  }
  
  gcc -pthread -B /usr/local/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -c _configtest.c -o _configtest.o
  failure.
  removing: _configtest.c _configtest.o
  compiling '_configtest.c':
  
  #ifndef __cplusplus
  static inline int static_func (void)
  {
      return 0;
  }
  inline int nostatic_func (void)
  {
      return 0;
  }
  #endif
  int main(void) {
      int r1 = static_func();
      int r2 = nostatic_func();
      return r1 + r2;
  }
  
  gcc -pthread -B /usr/local/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -c _configtest.c -o _configtest.o
  failure.
  removing: _configtest.c _configtest.o
  compiling '_configtest.c':
  
  #ifndef __cplusplus
  static __inline__ int static_func (void)
  {
      return 0;
  }
  __inline__ int nostatic_func (void)
  {
      return 0;
  }
  #endif
  int main(void) {
      int r1 = static_func();
      int r2 = nostatic_func();
      return r1 + r2;
  }
  
  gcc -pthread -B /usr/local/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -c _configtest.c -o _configtest.o
  failure.
  removing: _configtest.c _configtest.o
  compiling '_configtest.c':
  
  #ifndef __cplusplus
  static __inline int static_func (void)
  {
      return 0;
  }
  __inline int nostatic_func (void)
  {
      return 0;
  }
  #endif
  int main(void) {
      int r1 = static_func();
      int r2 = nostatic_func();
      return r1 + r2;
  }
  
  gcc -pthread -B /usr/local/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -c _configtest.c -o _configtest.o
  failure.
  removing: _configtest.c _configtest.o
  building 'bottleneck.reduce' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/bottleneck
  creating build/temp.linux-x86_64-3.6/bottleneck/src
  gcc -pthread -B /usr/local/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/tmp/pip-build-env-mg3llph8/overlay/lib/python3.6/site-packages/numpy/core/include -I/usr/local/include/python3.6m -Ibottleneck/src -c bottleneck/src/reduce.c -o build/temp.linux-x86_64-3.6/bottleneck/src/reduce.o -O2
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for bottleneck
Failed to build bottleneck
ERROR: Could not build wheels for bottleneck which use PEP 517 and cannot be installed directly
# 
</details>

**Comment by denphi** [comment]
This seems to be related to:  https://github.com/pydata/bottleneck/issues/281. gcc is not installed on that container.

I solved as suggested there, 8ac2d319e7ed is my docker instance ID,


```
docker exec -it -u root 8ac2d319e7ed /bin/bash
# apt-get update
# apt-get install gcc
```

**Comment by LukaPitamic** [comment]
wow, awesome!!! Thank you @denphi
